contracts/RewardDelegate.sol

```diff
@@ -23,6 +23,9 @@ contract RewardDelegate is IRewardDelegate {
     function setBeneficiaryCandidate(address candidate) external override {
         address truster = msg.sender;
 
+        // RD_TNC: truster is not a contract
+        require(truster.isContract(), "RD_TNC");
```

Comment:
is it necessary? What if I have 100 EOAs and want to set all of their beneficiary to one of them?

Reply:
per discussion, will limit the use to contract only